clang/test/CodeGenSYCL/intel-fpga-no-global-work-offset.cpp

```diff
@@ -1,4 +1,4 @@
-// RUN: %clang_cc1 -fsycl-is-device -internal-isystem %S/Inputs -triple spir64-unknown-unknown -disable-llvm-passes -sycl-std=2017 -emit-llvm -o - %s | FileCheck %s
+// RUN: %clang_cc1 -fsycl-is-device -internal-isystem %S/Inputs -triple spir64-unknown-unknown -disable-llvm-passes -sycl-std=2020 -emit-llvm -o - %s | FileCheck %s
```

Comment:
`-sycl-std=2020` is the default and the only supported version, so there is reason to set it explicitly. Right?

Reply:
> `-sycl-std=2020` is the default and the only supported version, so there is reason to set it explicitly. Right?
Then I'd like us to agree if all tests are using default version (except some specific) or vise-versa

Because, what will happen with all these clang/tests changed as you proposed in case of the next standard will be defaulted after 2020 version (eg 2024/2025/etc)?

Reply:
> Then I'd like us to agree:
> all tests have to use the default version (except some specific)

As far as I can tell, this is the way tests written today. This patch introduces exceptions to this rule and that's why I raised the question.

Reply:
> this is the way tests written today
> and that's why I raised the question.


Thank you for correcting my point of view, I see it now

```
find -path "*/test/*.cpp" -type f|xargs -I{} grep -anH '// RUN: %clang_cc1' {}|grep "\-sycl\-std="|awk '{print $1}'|wc -l
139

find -path "*/test/*.cpp" -type f|xargs -I{} grep -anH '// RUN: %clang_cc1' {}|grep -v "\-sycl\-std="|awk '{print $1}'|wc -l
23838
```

Reply:
> this is the way tests written today
> and that's why I raised the question.


Thank you for correcting my point of view, I see it now

```
find -path "*/test/*.cpp" -type f|xargs -I{} grep -anH '// RUN: %clang_cc1' {}|grep "\-sycl\-std="|awk '{print $1}'|wc -l
139

find -path "*/test/*.cpp" -type f|xargs -I{} grep -anH '// RUN: %clang_cc1' {}|grep -v "\-sycl\-std="|awk '{print $1}'|wc -l
23838
```

So as per your proposal the 's/(// RUN:.*) -sycl-std=\d+ (.*)/\1 \2/' is O.K.
But do you think that '// RUN.*(CHECK-SYCL(-STD-)?|-Wno-sycl-|-verify=sycl-)2020' should be handled as well?
There are 9 cases which I'm not sure how to resolve yet.

I'd move discussed changes to separate PR to not overload/mix this one. Are you O.K. with it?

Reply:
> So as per your proposal the 's/(// RUN:._) -sycl-std=\d+ (._)/\1 \2/' is O.K.

It's not okay. As you noted there are exceptions i.e. tests which validate `-sycl-std=2020` option.

My request is to go over the tests where you changed `-sycl-std=2017` to `-sycl-std=2020` and consider dropping the option altogether. 

Regarding separate PR. I don't see value in making these changes in separate commits. I prefer making the final change in this PR.